scripts/nasa_systems_check.sh

```
@@ -0,0 +1,595 @@
+        fi
+    done
+}
+
+# check_environment_variables verifies the presence of a .env file and checks that all critical environment variables required for the application are set.
+check_environment_variables() {
+    print_header "SYSTEMS CHECK 4: ENVIRONMENT VARIABLES VERIFICATION"
+    
+    # Load environment variables if .env exists
+    if [ -f ".env" ]; then
+        print_success ".env file exists"
+        export $(grep -v '^#' .env | xargs)
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)